### ISSUE TITLE
[8.1.17] Cache-Rebuild fails with Segmentation Fault.

### ISSUE THREAD

**Issue by showardclark** [drush-ops/drush#4145]
**Describe the bug**
When running the cache rebuild command ```drush cr```, the output reads ```segmentation fault``` and doesn't do anything.

**To Reproduce**
Ran ```drush cr```

**Expected behavior**
[success] Cache rebuilt

**Actual behavior**
Output just reads ```segmentation fault```

**Workaround**
Within the Drupal 8 UI, I can clear individual caches barre the routing and links cache - which fails.

### System Configuration
| Q               | A
| --------------- | ---
| Drush version?  | 8.1.17
| Drupal version? | 8.6.15
| PHP version     | 7.2
| OS?             | Linux (Ubuntu 18.04)

**Additional information**
I've been trying to set up a site locally and have run into issues with the drush cr command that I haven't experienced before. This is within drupalVM. I did have a prior issue where drush cr would take 6-7mins (I timed it with ```time drush cr```) and time out on memory, but that issue has since been solved by using a coworkers config and making a new VM.

DrupalVM version 2.02.
Vagrant version 2.0.3
VirtualBox 5.2.32
Ansible 2.5.1

Running with debug gives the following output:

```Cache HIT cid: 8.1.17-commandfiles-0-b5e951c05279061ad957a2bd0eba1f95 [0.01 sec, 3.55 MB]         [debug]
Begin redispatch via drush_invoke_process(). [0.1 sec, 8.15 MB]                                  [notice]
Backend invoke: env COLUMNS=105  /var/www/mysite/vendor/drush/drush/drush.launcher             [command]
--root=/var/www/mysite/web --debug --verbose  cache-rebuild 2>&1 [0.11 sec, 8.17 MB]
Calling proc_open(env COLUMNS=105  /var/www/mysite/vendor/drush/drush/drush.launcher  --root=/var/www/mysite/web --debug --verbose  cache-rebuild 2>&1);
Drush preflight prepare loaded autoloader at                         [preflight]
/var/www/mysite/vendor/autoload.php [0 sec, 3.18 MB]
Starting Drush preflight. [0 sec, 3.18 MB]                                                    [preflight]
Cache HIT cid: 8.1.17-commandfiles-0-b5e951c05279061ad957a2bd0eba1f95 [0.01 sec, 3.48 MB]         [debug]
Scanning into /etc/drush for /.*aliases\.drush(8|)rc\.php$/ [0.03 sec, 6.09 MB]                   [debug]
Scanning into /etc/drush for /self\.alias\.drush(8|)rc\.php$/ [0.03 sec, 6.09 MB]                 [debug]
Scanning into /var/www/mysite/vendor/drush/drush/includes/.. for                                 [debug]
/.*aliases\.drush(8|)rc\.php$/ [0.03 sec, 6.09 MB]
Scanning into /var/www/mysite/vendor/drush/drush/includes/.. for                                 [debug]
/self\.alias\.drush(8|)rc\.php$/ [0.04 sec, 6.09 MB]
Scanning into /home/vagrant/.drush for /.*aliases\.drush(8|)rc\.php$/ [0.04 sec, 6.09 MB]         [debug]
Scanning into /home/vagrant/.drush for /self\.alias\.drush(8|)rc\.php$/ [0.04 sec, 6.09 MB]       [debug]
Scanning into /var/www/mysite/web/../drush for /.*aliases\.drush(8|)rc\.php$/ [0.04 sec, 6.09    [debug]
MB]
Scanning into /var/www/mysite/web/../drush for /self\.alias\.drush(8|)rc\.php$/ [0.04 sec,       [debug]
6.09 MB]
Scanning into /var/www/mysite/web/drush for /.*aliases\.drush(8|)rc\.php$/ [0.04 sec, 6.1 MB]    [debug]
Scanning into /var/www/mysite/web/drush for /self\.alias\.drush(8|)rc\.php$/ [0.04 sec, 6.1      [debug]
MB]
Scanning into /var/www/mysite/web/sites/all/drush for /.*aliases\.drush(8|)rc\.php$/ [0.05       [debug]
sec, 6.1 MB]
Scanning into /var/www/mysite/web/sites/all/drush for /self\.alias\.drush(8|)rc\.php$/ [0.05     [debug]
sec, 6.1 MB]
Scanning into /var/www/mysite/web/sites/default for /.*aliases\.drush(8|)rc\.php$/ [0.05 sec,    [debug]
6.1 MB]
Scanning into /var/www/mysite/web/sites/default for /self\.alias\.drush(8|)rc\.php$/ [0.05       [debug]
sec, 6.1 MB]
Bootstrap to phase 0. [0.07 sec, 8.08 MB]                                                     [bootstrap]
Bootstrap to phase 2. [0.07 sec, 8.08 MB]                                                     [bootstrap]
Drush bootstrap phase : bootstrap_drupal_root() [0.08 sec, 8.21 MB]                           [bootstrap]
Initialized Drupal 8.6.15 root directory at /var/www/mysite/web [0.08 sec, 8.22 MB]          [bootstrap]
Find command files for phase 1 (max=2) [0.08 sec, 7.02 MB]                                        [debug]
Cache HIT cid: 8.1.17-commandfiles-1-bb8c2e5515a624b141d6ed683c214f0c [0.08 sec, 7.02 MB]         [debug]
Cache HIT cid: 8.1.17-annotationfiles-1-6506633595e55a61fa67e786a98c10d0 [0.08 sec, 7.1 MB]       [debug]
Scanning into /etc/drush for /.*aliases\.drush(8|)rc\.php$/ [0.08 sec, 7.1 MB]                    [debug]
Scanning into /etc/drush for /self\.alias\.drush(8|)rc\.php$/ [0.08 sec, 7.1 MB]                  [debug]
Scanning into /var/www/mysite/vendor/drush/drush/includes/.. for                                 [debug]
/.*aliases\.drush(8|)rc\.php$/ [0.08 sec, 7.1 MB]
Scanning into /var/www/mysite/vendor/drush/drush/includes/.. for                                 [debug]
/self\.alias\.drush(8|)rc\.php$/ [0.09 sec, 7.1 MB]
Scanning into /home/vagrant/.drush for /.*aliases\.drush(8|)rc\.php$/ [0.09 sec, 7.1 MB]          [debug]
Scanning into /home/vagrant/.drush for /self\.alias\.drush(8|)rc\.php$/ [0.09 sec, 7.1 MB]        [debug]
Scanning into /var/www/mysite/web/../drush for /.*aliases\.drush(8|)rc\.php$/ [0.09 sec, 7.1     [debug]
MB]
Scanning into /var/www/mysite/web/../drush for /self\.alias\.drush(8|)rc\.php$/ [0.09 sec,       [debug]
7.1 MB]
Scanning into /var/www/mysite/web/drush for /.*aliases\.drush(8|)rc\.php$/ [0.09 sec, 7.1 MB]    [debug]
Scanning into /var/www/mysite/web/drush for /self\.alias\.drush(8|)rc\.php$/ [0.09 sec, 7.11     [debug]
MB]
Scanning into /var/www/mysite/web/sites/all/drush for /.*aliases\.drush(8|)rc\.php$/ [0.09       [debug]
sec, 7.11 MB]
Scanning into /var/www/mysite/web/sites/all/drush for /self\.alias\.drush(8|)rc\.php$/ [0.09     [debug]
sec, 7.11 MB]
Scanning into /var/www/mysite/web/sites/default for /.*aliases\.drush(8|)rc\.php$/ [0.1 sec,     [debug]
7.11 MB]
Scanning into /var/www/mysite/web/sites/default for /self\.alias\.drush(8|)rc\.php$/ [0.1        [debug]
sec, 7.11 MB]
Drush bootstrap phase : bootstrap_drupal_site() [0.1 sec, 7.63 MB]                            [bootstrap]
Initialized Drupal site default at sites/default [0.1 sec, 7.63 MB]                           [bootstrap]
Find command files for phase 2 (max=2) [0.11 sec, 7.63 MB]                                        [debug]
Cache HIT cid: 8.1.17-install_profile-66ecfeb9791a023150773849f1550c5d [0.11 sec, 7.63 MB]        [debug]
Cache MISS cid: 8.1.17-commandfiles-2-bbdc55ee41d2d85d181b6a7385fc8c3c [0.11 sec, 7.63 MB]        [debug]
Cache SET cid: 8.1.17-commandfiles-2-bbdc55ee41d2d85d181b6a7385fc8c3c [1.19 sec, 7.64 MB]         [debug]
Found command: cache-rebuild (commandfile=cache) [1.2 sec, 8.3 MB]                            [bootstrap]
Calling hook drush_cache_rebuild [1.22 sec, 10.54 MB]                                             [debug]
Segmentation fault
End redispatch via drush_invoke_process(). [20.7 sec, 8.17 MB]  ```

**Comment by weitzman** [comment]
Most likely something other than Drush is causing the fault. That is, code that contrib or core drupal code that gets called during any cache rebuild. Consider looking at the cli php error log.

**Comment by showardclark** [comment]
Both my php_errors.log and php-fpm.log files shows nothing other than test errors or unrelated messages, is there somewhere else I should be looking @weitzman?

**Comment by greg-1-anderson** [comment]
It's really unusual for php code to cause a segfault. It's more likely a php extension, if you added or upgraded one of those recently. Could be something that went wrong in the system, unrelated to PHP. You could try running the same site on a different server / VM and see if it still segfaults.

The error log will rarely tell you anything about a segfault. You need to capture the core dump and get a stack trace. This will not always yield interesting results.

**Comment by showardclark** [comment]
I've destroyed and remade the VM a few times with various different vagrant files and configs and all of them have some level of issue with running a drush cr. I've narrowed the issue down to happening purely when clearing the router cache. 

I had some luck with making it work briefly by disabling xdebug, but it has since stopped working even after I made no further changes to the set up of the VM.
